src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,72 @@
+package mate.academy.service.impl;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+
+    private static final String HASHING_ALGORITHM = "SHA-256";
+    @Inject
+    private final UserService userService;
```

Comment:
The `@Inject` annotation is used on a final field `userService`. This is not compatible with most dependency injection frameworks, as they require a no-arg constructor or setter methods to inject dependencies. Consider removing the `final` keyword or using constructor injection without the `@Inject` annotation.